python/MyriaPythonWorker.py

```diff
@@ -0,0 +1,215 @@
+            self.write_int(obj, stream)
+        elif output_type == DataType.LONG:
+            self.write_int(DataType.LONG, stream)
+            self.write_long(obj, stream)
+        elif output_type == DataType.FLOAT:
+            self.write_int(DataType.FLOAT, stream)
+            self.write_float(obj, stream)
+        elif output_type == DataType.DOUBLE:
+            self.write_int(DataType.DOUBLE, stream)
+            self.write_double(obj, stream)
+        elif output_type == DataType.BLOB:
+            self.write_int(DataType.BLOB, stream)
+            self.pickle_and_write(self, obj, stream)
+
+    @classmethod
+    def read_command(self, stream):
+        length = self.read_int(stream)
+
+        if length < 0:
+            raise ValueError("Command cannot be less than zero.")
+        s = stream.read(length)
+
+        if len(s) < length:
+            raise EOFError
+        unenc = base64.urlsafe_b64decode(s)
+        return self.loads(unenc)
+
+    @staticmethod
+    def dumps(obj):
+        protocol = 2
+        return cPickle.dumps(obj, protocol)
+
+    @staticmethod
+    def loads(obj):
+        return cPickle.loads(obj)
+
+    @staticmethod
+    def pickle_and_write(self, obj, stream):
+        serialized = self.dumps( obj)
+
+        if serialized is None:
+            raise ValueError("Serialized value should not be None.")
+        elif len(serialized) > (1 << 31):
+            raise ValueError("Cannot serialize object larger than 2G.")
+
+        self.write_int(len(serialized), stream)
+        stream.write(serialized)
+
+
+def main(in_file, out_file):
+    pickle_serializer = PickleSerializer()
+    try:
+        func = pickle_serializer.read_command(in_file)
+        tuple_size = pickle_serializer.read_int(in_file)
+        output_type = pickle_serializer.read_int(in_file)
+        is_flatmap = pickle_serializer.read_int(in_file)
+
+        if tuple_size < 1:
+            raise ValueError("size of tuple should not be less than 1 ")
+
+        while True:
+            num_tuples = pickle_serializer.read_int(in_file)
+            if num_tuples == SpecialLengths.END_OF_STREAM:
+                exit(0)
+
+            tuple_list = []
+            for j in range(num_tuples):
+                tuple_list.append(pickle_serializer.read_tuple(in_file, tuple_size))
+
+            retval = func(tuple_list)
+            if is_flatmap > 0 :
+                count = len(retval)
+                pickle_serializer.write_int(count, out_file)
+                for i in range(count):
+                    pickle_serializer.write_with_length(
+                        retval[i], out_file, output_type)
+            else:
+                pickle_serializer.write_with_length(retval, out_file, output_type)
+
+            out_file.flush()
+
+    except Exception:
+        try:
+            pickle_serializer.write_int(SpecialLengths.PYTHON_EXCEPTION_THROWN, out_file)
+            pickle_serializer.write_with_length(traceback.format_exc().encode(
```

Comment:
Use the `DataType` value, not the magic number `5`, so I don't have to look it up to know what it means.